From d3164ae64c7cf108e0c304e5a1a7a3383ec994fd Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Mon, 14 Sep 2020 21:02:43 +0800
Subject: [PATCH] rtc: pcf8563: make the 32k clock output to be always enabled

The WiFi needs a always-on 32.768k clock on rv1126-battery-v10 board.
The original driver disables the clock whilst registering the
*rtc_clk*, we have to avoid *disable* to make sure WiFi is available.

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
Change-Id: Ia612b17a3d9fdef2a3d37ca2862c46b81db9133d
---
 drivers/rtc/rtc-pcf8563.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/rtc/rtc-pcf8563.c b/drivers/rtc/rtc-pcf8563.c
index d8adf69b6697..5092452c57e8 100644
--- a/drivers/rtc/rtc-pcf8563.c
+++ b/drivers/rtc/rtc-pcf8563.c
@@ -513,7 +513,7 @@ static const struct clk_ops pcf8563_clkout_ops = {
 	.set_rate = pcf8563_clkout_set_rate,
 };
 
-static struct clk *pcf8563_clkout_register_clk(struct pcf8563 *pcf8563)
+static __maybe_unused struct clk *pcf8563_clkout_register_clk(struct pcf8563 *pcf8563)
 {
 	struct i2c_client *client = pcf8563->client;
 	struct device_node *node = client->dev.of_node;
@@ -614,7 +614,7 @@ static int pcf8563_probe(struct i2c_client *client,
 
 	}
 
-#ifdef CONFIG_COMMON_CLK
+#if defined(CONFIG_COMMON_CLK) && !defined(CONFIG_ROCKCHIP_THUNDER_BOOT)
 	/* register clk in common clk framework */
 	pcf8563_clkout_register_clk(pcf8563);
 #endif
-- 
2.35.3

